api/src/main/java/org/openmrs/module/commonreports/reports/AntenatalReportManager.java

```diff
@@ -0,0 +1,378 @@
+		malariaPositiveConceptSet.add(inizService.getConceptFromKey("report.antenatal.twoPlus"));
+		malariaPositiveConceptSet.add(inizService.getConceptFromKey("report.antenatal.threePlus"));
+		malariaPositiveConceptSet.add(inizService.getConceptFromKey("report.antenatal.fourPlus"));
+		
+		malaria.setValueList(malariaPositiveConceptSet);
+		
+		SqlCohortDefinition sql = new SqlCohortDefinition(
+		        "select patient_id from orders where concept_id=(select DISTINCT concept_id from concept where uuid = '"
+		                + inizService.getValueFromKey("report.antenatal.chloroquine")
+		                + "' and retired = 0) and order_type_id =(select order_type_id from order_type where uuid = '"
+		                + inizService.getValueFromKey("report.antenatal.drugOrder") + "')");
+		sql.addParameter(new Parameter("onOrAfter", "On Or After", Date.class));
+		sql.addParameter(new Parameter("onOrBefore", "On Or Before", Date.class));
+		CompositionCohortDefinition ccd2 = new CompositionCohortDefinition();
+		ccd2.initializeFromElements(_prenatal, sql, malaria, female);
+		antenatalRisks.addRow(
+		    MessageUtil.translate("commonreports.report.antenatalRisks.prenatalMalariaPositiveChloroquine"), ccd2,
+		    parameterMappings);
+		
+		//Prenatal + MUAC =<21cm
+		NumericObsCohortDefinition muac = new NumericObsCohortDefinition();
+		muac.addParameter(new Parameter("onOrAfter", "On Or After", Date.class));
+		muac.addParameter(new Parameter("onOrBefore", "On Or Before", Date.class));
+		muac.setQuestion(inizService.getConceptFromKey("report.antenatal.midUpperArmCircumference"));
+		muac.setOperator1(RangeComparator.GREATER_EQUAL);
+		muac.setValue1(0.0);
+		muac.setOperator2(RangeComparator.LESS_EQUAL);
+		muac.setValue2(210.0);
+		CompositionCohortDefinition ccd3 = new CompositionCohortDefinition();
+		ccd3.initializeFromElements(_prenatal, muac, female);
+		antenatalRisks.addRow(MessageUtil.translate("commonreports.report.antenatalRisks.prenatalMUAC=<21cm"), ccd3,
+		    parameterMappings);
+		
+		//Women + fer folate co prescribed 
+		VisitService vs = Context.getVisitService();
+		List<VisitType> lVT = new ArrayList<VisitType>();
+		String[] otherVisitTypes = inizService.getValueFromKey("report.antenatal.otherVisitTypes").split(",");
+		
+		for (String member : otherVisitTypes) {
+			lVT.add(vs.getVisitTypeByUuid(member));
+		}
+		
+		VisitCohortDefinition _other = new VisitCohortDefinition();
+		_other.setVisitTypeList(lVT);
+		
+		SqlCohortDefinition sqdc = new SqlCohortDefinition(
+		        "select patient_id from orders where concept_id=(select DISTINCT concept_id from concept where uuid = '"
+		                + inizService.getValueFromKey("report.antenatal.ferrousFolate")
+		                + "' and retired = 0) and order_type_id =(select order_type_id from order_type where uuid = '"
+		                + inizService.getValueFromKey("report.antenatal.drugOrder") + "')");
```

Comment:
Let's therefore add something like `... AND date_created BETWEEN :onOrAfter AND : onOrBefore` to the query.

Reply:
Isn't this handled in the following line 286

Reply:
Same here...

Reply:
Done